framework/doc/content/syntax/Mesh/index.md

```diff
@@ -165,4 +165,25 @@ recycled (because it might be a key to an important map), you should use unique_
 
 The MooseMesh object has a method for building a map (technically a multimap) of paired periodic nodes in the
 simulation. This map provides a quick lookup of all paired nodes on a periodic boundary. in the 2D and 3D cases
-each corner node will map to 2 or 3 other nodes (respectively).
\ No newline at end of file
+each corner node will map to 2 or 3 other nodes (respectively).
+
+## Extra integer IDs
+
+Extra integer IDs for all the elements of a mesh can be useful for handling complicated material assignment, performing specific calculations on groups of elements, etc.
+Often times, we do not want to use subdomain IDs for these tasks because otherwise too many subdomains could be needed, and in turn large penalty on run-time performance could be introduced.
+
+MooseMesh[MooseMesh.md] has a parameter `extra_integers` to allow users to introduce more integer IDs for elements each identified with a name in the parameter.
+When this parameter is specified, extra integers will be made available for all elements through `Assembly` in MOOSE objects such as kernels, aux kernels, materials, initial conditions, element user objects, etc.
+To retrieve the integer on an element, one needs to simply call
+```
+getElementID(integer_name_parameter, comp),
+```
+where `integer_name_parameter` is the name of the parameter in type of `std::vector<std::string>` of this object listing all integer names.
+`comp` is the number of the integer names the returned $_id$ is for.
+It is noticed that the returned value of this function call must be in type of `const dof_id_type &`, which is used to refer the value set by MOOSE in `Assembly`.
+Based on this ID, one can proceed with any particular operations, for example, choosing a different set of data for evaluating material properties.
+
+The IDs can be assigned to the mesh elements with mesh modifiers similarly as assigning subdomain IDs.
+Currently we only extended SubdomainBoundingBoxGenerator[SubdomainBoundingBoxGenerator.md] for such assignment.
```

Comment:
Let's remove all of this stuff about what has this capability and what might get it in the future.  We will forget to update this documentation as things change.